src/lib/y2firstboot/clients/hostname.rb

```diff
@@ -0,0 +1,197 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+Yast.import "UI"
+Yast.import "Arch"
+Yast.import "DNS"
+Yast.import "Host"
+Yast.import "NetworkConfig"
+Yast.import "String"
+Yast.import "Wizard"
+Yast.import "ProductControl"
+Yast.import "ProductFeatures"
+Yast.import "GetInstArgs"
+
+module Y2Firstboot
+  module Clients
+    # Client to set the hostname during first boot
+    #
+    # This is just a renamed version of InstHostnameClient, which was removed from
+    # YaST2-Network when the second stage was removed from the installation
+    # process
+    class Hostname < Yast::Client
+      class << self
+        # @!method valid_dns_proposal=(value)
+        #   @param [Boolean] Whether a valid DNS proposal was done
+        attr_writer :valid_dns_proposal
+
+        # Determines whether a valid DNS proposal was done
+        #
+        # @return [Boolean] Returns true if a DNS proposal was done
+        def valid_dns_proposal
+          @valid_dns_proposal ||= false
```

Comment:
btw as it is boolean, it is in fact same as `@valid_dns_proposal = false` or `!!@valid_dns_proposal`

Reply:
Yep, to be honest just reused it from previous code, but was already thinking in remove it, an do the initialization when creating a new instance, but  I suppose the idea is to really set it when checked the very first time but do not see any use of it that could get a benefit from the current behavior

Reply:
I will leave as it is.